### DISCUSSION TITLE
`shebang` sub-command naming

### DISCUSSION THREAD

**Discussion by Gedochao** [VirtusLab/scala-cli#1755] [General]
> Can we do better than `shebang` on the name? It seems like obscure jargon to me

(answering https://github.com/VirtusLab/scala-cli/pull/1740#issuecomment-1372469594)
@SethTisue any suggestions for an alias?

I actually don't think it's that obscure of a jargon...
The current `shebang` sub-command has 2 main use cases:
- to be used in the `shebang` header, as per [the Unix standard](https://en.wikipedia.org/wiki/Shebang_(Unix)), for which the name seems appropriate;
- to be used from the command line with legacy syntax backwards compatible with the old `scala` runner.

Although I believe we'll be keeping the `shebang` name, as the former usecase is the main one, I think we could potentially have an alias to make it easier to guess the second one... provided that we can think of a good name for it, of course.
Would something like `run-legacy` do the trick? feels verbose to me...

Also, I wonder if an alias is even necessary... if a user cares about backwards compatibility in this context when migrating from the old `scala` runner to Scala CLI, then it's probably fair to assume he's a rare specimen of a power user. Which means he should probably be okay with obscurely named sub-commands...

We don't want to promote using `shebang` from the command line for regular users, they should just learn the `run` syntax.

**Comment by Flowdalic** [comment]
Does it have to be a subcommand? Can't scala-cli not inspect the name under it was called and act accordingly? Then you could have e.g.,

```
#!/usr/bin/env scala-script
```

which looks nice, clean and idiomatic. And `/usr/bin/scala-script` is a symlink to `/usr/bin/scala-cli`.

**Reply by Gedochao** [reply]
That is actually how we've been differentiating between SIP-mode and the standard `scala-cli`... well, we still do, until #1824 is resolved (a solution is already in-progress in #1849)

Regardless, I believe we do still want `shebang` available as a sub-command, as some people prefer its command line behaviour over `run`. Not to mention it's easier to document it as a sub-command.
`shebang` sort of mimics how the old `scala` runner treated program arguments (all args started with the second are treated as program args rather than inputs, no need for `--`) and apparently some people are used to handling it this way.
